runtime/lua/vim/treesitter/_range.lua

```diff
@@ -2,23 +2,23 @@ local api = vim.api
 
 local M = {}
 
----@class Range2
----@field [1] integer start row
----@field [2] integer end row
-
----@class Range4
----@field [1] integer start row
----@field [2] integer start column
----@field [3] integer end row
----@field [4] integer end column
-
----@class Range6
----@field [1] integer start row
----@field [2] integer start column
----@field [3] integer start bytes
----@field [4] integer end row
----@field [5] integer end column
----@field [6] integer end bytes
+---@class Range2 Range represented by rows
+---@field [1] integer start row (0-indexed).
+---@field [2] integer end row (0-indexed, exclusive).
```

Comment:
Is end row really exclusive?, or is it just end col and end byte

Reply:
I think the relevant usage is
```
    local tree = parser:parse({ start_row, end_row })
```

Also there is a trivial upcasting from Range2 to Range4:
```
(3, 7) => (3, 0, 7, 0)
```
which represents the exactly the same region. `(3, 0, 7, 0)` might be equivalent to `(3, 0, 6, 80)` assuming the number of columns is 80 in line 6 (0-indexed).